Import from hub docs

nataliaElv opened the pull request:
# Description
<!-- Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change. -->

Closes #<issue_number>

**Type of change**
<!--  Please delete options that are not relevant. Remember to title the PR according to the type of change  -->

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Refactor (change restructuring the codebase without changing functionality)
- Improvement (change adding some improvement to an existing functionality)
- Documentation update

**How Has This Been Tested**
<!--  Please add some reference about how your feature has been tested.  -->

**Checklist**
<!--  Please go over the list and make sure you've taken everything into account -->

- I added relevant documentation
- I followed the style guidelines of this project
- I did a self-review of my code
- I made corresponding changes to the documentation
- I confirm My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

leiyre committed:
prevent drag when the input is focused

leiyre committed:
create questions

leiyre committed:
show metadata types and allow to change

leiyre committed:
update error message

jfcalvo committed:
feat: change dataset publish validation from at least one required field to at least one field (required or not) (#5569)

# Description

In this PR we are changing how dataset publish validation works moving
from:
* At least one required field to be a valid publishable dataset.

To:

* At least one field (required or not) to be a valid publishable
dataset.

Refs https://github.com/argilla-io/roadmap/issues/21

**Type of change**

- New feature (non-breaking change which adds functionality)


**How Has This Been Tested**

- [x] Modifying test suite. 

**Checklist**

- I added relevant documentation
- I followed the style guidelines of this project
- I did a self-review of my code
- I made corresponding changes to the documentation
- I confirm My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature
works
- I have added relevant notes to the CHANGELOG.md file (See
https://keepachangelog.com/)

jfcalvo committed:
chore: Update CHANGELOG.md

frascuchon committed:
[Refactor] Remove settings name pattern restrictions (#5573)

# Description
<!-- Please include a summary of the changes and the related issue.
Please also include relevant motivation and context. List any
dependencies that are required for this change. -->

This PR removes name pattern validation for field, question,
metadata-property and vector-settings

**Type of change**
<!-- Please delete options that are not relevant. Remember to title the
PR according to the type of change -->

- Refactor (change restructuring the codebase without changing
functionality)
- Improvement (change adding some improvement to an existing
functionality)

**How Has This Been Tested**
<!-- Please add some reference about how your feature has been tested.
-->

**Checklist**
<!-- Please go over the list and make sure you've taken everything into
account -->

- I added relevant documentation
- I followed the style guidelines of this project
- I did a self-review of my code
- I made corresponding changes to the documentation
- I confirm My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature
works
- I have added relevant notes to the CHANGELOG.md file (See
https://keepachangelog.com/)

frascuchon committed:
Merge branch 'develop' into feat/argilla-direct-feature-branch

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks
for more information, see https://pre-commit.ci

leiyre committed:
update repository Id example

leiyre committed:
include metadata and restructure components

leiyre committed:
metadata type

leiyre committed:
update some styles

leiyre committed:
fix validation

leiyre committed:
remove console

leiyre committed:
fix Mtetadata creation class

damianpumar committed:
feat: Improve design

leiyre committed:
Merge branch 'feat/create-dataset-from-ui' of https://github.com/argilla-io/argilla into feat/create-dataset-from-ui

leiyre committed:
fix param value

damianpumar committed:
feat: :art: Improve design

damianpumar committed:
feat: :art: Fix lint

leiyre committed:
update title

leiyre committed:
update question  prefilled settings

leiyre committed:
initialize answers for preview

leiyre committed:
Merge branch 'feat/create-dataset-from-ui' of https://github.com/argilla-io/argilla into feat/create-dataset-from-ui

leiyre committed:
fix subset selector when single subset

leiyre committed:
fix draggable areas

leiyre committed:
refactor

leiyre committed:
allow to choose column from question card

leiyre committed:
refresh static settings data on changing type

leiyre committed:
fix condition

leiyre committed:
update structure

leiyre committed:
refactor DatasetConfigurationColumnSelector to DatasetConfigurationSelector

leiyre committed:
show default setting when no mapping is selected

leiyre committed:
include dataset configuration name and workspace

leiyre committed:
create prop for rezisable minHeightPercent

leiyre committed:
Merge branch 'develop' into feat/create-dataset-from-ui

damianpumar committed:
🐛 Fix

leiyre committed:
fix lint

leiyre committed:
fix comparison error

leiyre committed:
remove default question and field

leiyre committed:
explicit import icons

leiyre committed:
include resize between preview and form

leiyre committed:
include dtype in FieldCreation

leiyre committed:
remove metadata references in templates

leiyre committed:
fix test

leiyre committed:
remove markdown and titile edition

leiyre committed:
include required

damianpumar committed:
feat: :art: Improve

damianpumar committed:
Merge branch 'feat/create-dataset-from-ui' of https://github.com/argilla-io/argilla into feat/create-dataset-from-ui

leiyre committed:
Merge branch 'feat/create-dataset-from-ui' of https://github.com/argilla-io/argilla into feat/create-dataset-from-ui

damianpumar committed:
✨ Fix

damianpumar committed:
🔥 Fix

leiyre committed:
show column type in cards

leiyre committed:
include mocked field preview

leiyre committed:
get types for field preview

leiyre committed:
Merge branch 'feat/create-dataset-from-ui' of https://github.com/argilla-io/argilla into feat/create-dataset-from-ui

leiyre committed:
fix resize id

leiyre committed:
update generated question name

leiyre committed:
include padding

damianpumar committed:
🐛 Add workspace creation

damianpumar committed:
Merge branch 'feat/create-dataset-from-ui' of https://github.com/argilla-io/argilla into feat/create-dataset-from-ui

damianpumar committed:
🚑 Fix DI

damianpumar committed:
🚑 Get first record

damianpumar committed:
🚑 Rename

damianpumar committed:
✨ Fix endpoint

damianpumar committed:
💄 Send repo id instead of name

damianpumar committed:
🚑 Fix errors

damianpumar committed:
🚀 Fix

frascuchon committed:
Merge branch 'develop' into feat/argilla-direct-feature-branch

frascuchon committed:
chore: Fix tests after merge

frascuchon committed:
Merge branch 'develop' into feat/create-dataset-from-ui

frascuchon committed:
Merge branch 'feat/argilla-direct-feature-branch' into feat/create-dataset-from-ui

damianpumar committed:
🚑 Improve

damianpumar committed:
✨ Improve

damianpumar committed:
🔥 Add metadata

damianpumar committed:
🚑 Improve connection

damianpumar committed:
✨ Improve code

damianpumar committed:
✨ Fix lint

damianpumar committed:
🚀 Improve

jfcalvo committed:
feat: Add `metadata` support for datasets (#5586)

# Description

This PR add a new `metadata` column to `datasets` table and the
following changes:
* When creating a dataset now it is possible to specify `metadata`
values.
* When updating a dataset now it is possible to specify `metadata`
values.
* Now `metadata` attribute is included on `Dataset` schema so the
`metadata` of a dataset is exposed in our API.

Refs https://github.com/argilla-io/roadmap/issues/21

**Type of change**

- New feature (non-breaking change which adds functionality)

**How Has This Been Tested**

- [x] Adding new tests.

**Checklist**

- I added relevant documentation
- I followed the style guidelines of this project
- I did a self-review of my code
- I made corresponding changes to the documentation
- I confirm My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature
works
- I have added relevant notes to the CHANGELOG.md file (See
https://keepachangelog.com/)

---------

Co-authored-by: Paco Aranda <frascuchon@gmail.com>

frascuchon committed:
Merge branch 'feat/argilla-direct-feature-branch' into feat/create-dataset-from-ui

damianpumar committed:
✨ FIX, wip unit test...

damianpumar committed:
Merge branch 'feat/create-dataset-from-ui' of https://github.com/argilla-io/argilla into feat/create-dataset-from-ui

damianpumar committed:
⚡️ Fix metadata and improve mapping

damianpumar committed:
🐛 Fix, review with Leire

damianpumar committed:
⚡️ Fix

leiyre committed:
Merge branch 'feat/create-dataset-from-ui' of https://github.com/argilla-io/argilla into feat/create-dataset-from-ui

leiyre committed:
update UI actions structure

damianpumar committed:
💄 Improve dynamic fields

damianpumar committed:
⚡️ Fix nuxt

leiyre committed:
update UI

leiyre committed:
Merge branch 'feat/create-dataset-from-ui' of https://github.com/argilla-io/argilla into feat/create-dataset-from-ui

damianpumar committed:
🔥 Remove visible options

damianpumar committed:
Merge branch 'feat/create-dataset-from-ui' of https://github.com/argilla-io/argilla into feat/create-dataset-from-ui

frascuchon committed:
Merge branch 'develop' into feat/argilla-direct-feature-branch

damianpumar committed:
✨ Fix question

damianpumar committed:
✨ Redirect when the dataset has records

leiyre committed:
column selector

leiyre committed:
Merge branch 'feat/create-dataset-from-ui' of https://github.com/argilla-io/argilla into feat/create-dataset-from-ui

leiyre committed:
update styles

leiyre committed:
update dialog

leiyre committed:
include intro before dropdown option list

leiyre committed:
fix lint

leiyre committed:
fix test

leiyre committed:
update chips

leiyre committed:
fix chips selector reactivity

leiyre committed:
add isLoading prop

leiyre committed:
update styles

leiyre committed:
lint fix

leiyre committed:
include chips translations and update styles

leiyre committed:
remove hover

leiyre committed:
update styles

leiyre committed:
update static data

leiyre committed:
restructure

leiyre committed:
update styles

leiyre committed:
update styles

leiyre committed:
update padding

damianpumar committed:
✨ Improve navigation

damianpumar committed:
Merge branch 'feat/create-dataset-from-ui' of https://github.com/argilla-io/argilla into feat/create-dataset-from-ui

jfcalvo committed:
Merge branch 'develop' into feat/argilla-direct-feature-branch

leiyre committed:
update layout and styles

leiyre committed:
Merge branch 'feat/create-dataset-from-ui' of https://github.com/argilla-io/argilla into feat/create-dataset-from-ui

leiyre committed:
Update styles and copy

leiyre committed:
update border color

leiyre committed:
remove default value for resize

damianpumar committed:
🔥 Little improves

leiyre committed:
prevent white spaces

leiyre committed:
show selected column and prevent map when some type are selected

leiyre committed:
Merge branch 'feat/create-dataset-from-ui' of https://github.com/argilla-io/argilla into feat/create-dataset-from-ui

jfcalvo committed:
feat: add server dataset hub import (#5591)

# Description

This PR add changes to support the import of datasets from HF Hub into
Argilla with the following features:
* Add a new `POST /api/v1/datasets/:dataset_id/import` endpoint that
will enqueue a background job to import records from a HF Hub dataset,
returning information about the enqueued job. It expect the following
parameters:
* `name`: the name of the dataset (i.e. `lhoestq/demo1`) @burtenshaw
suggested changing it to `repo_id`
* `subset`: the dataset subset (i.e. `default`) @burtenshaw suggested to
make the parameter optional
* `split`: the dataset split (i.e. `train`) @burtenshaw suggested to
make the parameter opcional
* Add a new background job so the import process can be done outside of
request time.
* Add a new `HubDataset` class encapsulating all the logic to import a
dataset from the Hub.
* Add a new `/api/v1/jobs/:job_id` to get information about the status
of one specific job. This is useful if the UI or the SDK needs to know
if the import process finished. (@frascuchon we can use this to give
information about other processes, for example when a dataset
distribution settings is changed).

Refs https://github.com/argilla-io/roadmap/issues/21

**Type of change**

- New feature (non-breaking change which adds functionality)

**How Has This Been Tested**

- [x] Manually testing and adding more automatic tests to our suite.

**Checklist**

- I added relevant documentation
- I followed the style guidelines of this project
- I did a self-review of my code
- I made corresponding changes to the documentation
- I confirm My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature
works
- I have added relevant notes to the CHANGELOG.md file (See
https://keepachangelog.com/)

---------

Co-authored-by: Paco Aranda <frascuchon@gmail.com>
Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

leiyre committed:
disable button when no workspace

frascuchon committed:
Update .github/workflows/argilla.yml

jfcalvo committed:
feat: fix importing datasets features mapped as chat fields (#5611)

# Description

Using the dataset
https://huggingface.co/datasets/mlabonne/ultrachat_200k_sft we have
found that the import feature was not mapping correctly the `message`
feature.

In order to fix this I'm improving with this PR how the feature values
casting is done, checking if the features are instances of certain
feature classes instead of using the `_type` method.

I have also added a new test importing the `mlabonne/ultrachat_200k_sft`
dataset and using chat fields.

Refs https://github.com/argilla-io/roadmap/issues/21

**Type of change**

- New feature (non-breaking change which adds functionality)

**How Has This Been Tested**

- [x] Adding new tests to the suite.

**Checklist**

- I added relevant documentation
- I followed the style guidelines of this project
- I did a self-review of my code
- I made corresponding changes to the documentation
- I confirm My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature
works
- I have added relevant notes to the CHANGELOG.md file (See
https://keepachangelog.com/)

leiyre committed:
include split selector

leiyre committed:
allow images in preview

leiyre committed:
fix lint

damianpumar committed:
⚡️ Fix

damianpumar committed:
🔥 Add code snippet and improve fields

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks
for more information, see https://pre-commit.ci

leiyre committed:
include import data component

leiyre committed:
fix lint

leiyre committed:
update dataset mock

jfcalvo committed:
feat: add support to use datasets using features with sequences of class labels (#5613)

# Description

This PR adds the following changes:
* Add casting for features using sequences of class labels (casting them
using `int2str` function).
* Casting to string values for suggestions mapped to multi label
questions (iterating over the values).

Refs https://github.com/argilla-io/roadmap/issues/21

**Type of change**

- New feature (non-breaking change which adds functionality)

**How Has This Been Tested**

- [x] Adding additional tests to our suite.

**Checklist**

- I added relevant documentation
- I followed the style guidelines of this project
- I did a self-review of my code
- I made corresponding changes to the documentation
- I confirm My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature
works
- I have added relevant notes to the CHANGELOG.md file (See
https://keepachangelog.com/)

damianpumar committed:
🚑 Improve code

damianpumar committed:
🚑 Fix test

leiyre committed:
replace folder name for home

leiyre committed:
new icons

leiyre committed:
Merge branch 'feat/create-dataset-from-ui' of https://github.com/argilla-io/argilla into feat/create-dataset-from-ui

leiyre committed:
include new home page

leiyre committed:
temporal change

leiyre committed:
remove temporal page

leiyre committed:
fix lint and update styles

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks
for more information, see https://pre-commit.ci

leiyre committed:
update rating values

damianpumar committed:
✨ Improve question generation and add multi label

leiyre committed:
update snippet

leiyre committed:
Merge branch 'feat/create-dataset-from-ui' of https://github.com/argilla-io/argilla into feat/create-dataset-from-ui

leiyre committed:
update routes

leiyre committed:
lint fix

damianpumar committed:
Merge branch 'feat/create-dataset-from-ui' of https://github.com/argilla-io/argilla into feat/create-dataset-from-ui

damianpumar committed:
🚑 Fix map column

damianpumar committed:
✨ Fix axios

jfcalvo committed:
feat: add dataset split to be used along with row idx when `external_id` is not provided on mapping (#5616)

# Description

This PR add the dataset imported split to be used as `external_id` when
there is no value for `external_id` specified on the import mapping.

If importing the split `train` for a dataset and no `external_id` is
provided the `external_id` will be calculated like the following:
- `train_0`: first row of `train` split.
- `train_1`: second row of `train` split.
- ...

With this we are avoiding row duplications when another split is
imported to the same dataset. So if later we import the `test` split for
the same dataset we will have for `external_id`:
- `train_0`: first row of `train` split.
- `train_1`: second row of `train` split.
- ...
- `test_0`: first row of `test` split.
- `test_1`: second row of `test` split.
- ...

Refs https://github.com/argilla-io/roadmap/issues/21

**Type of change**

- New feature (non-breaking change which adds functionality)

**How Has This Been Tested**

- [x] Adding additional tests.

**Checklist**

- I added relevant documentation
- I followed the style guidelines of this project
- I did a self-review of my code
- I made corresponding changes to the documentation
- I confirm My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature
works
- I have added relevant notes to the CHANGELOG.md file (See
https://keepachangelog.com/)

frascuchon committed:
[REFACTOR] argilla: using split and row idx for default external (#5617)

# Description
<!-- Please include a summary of the changes and the related issue.
Please also include relevant motivation and context. List any
dependencies that are required for this change. -->

Using row idx + split name to generate default record id.


**Type of change**
<!-- Please delete options that are not relevant. Remember to title the
PR according to the type of change -->
- Refactor (change restructuring the codebase without changing
functionality)
- Improvement (change adding some improvement to an existing
functionality)

**How Has This Been Tested**
<!-- Please add some reference about how your feature has been tested.
-->

**Checklist**
<!-- Please go over the list and make sure you've taken everything into
account -->

- I added relevant documentation
- I followed the style guidelines of this project
- I did a self-review of my code
- I made corresponding changes to the documentation
- I confirm My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature
works
- I have added relevant notes to the CHANGELOG.md file (See
https://keepachangelog.com/)

frascuchon committed:
Merge branch 'develop' into feat/argilla-direct-feature-branch

damianpumar committed:
🚑 Fix

leiyre committed:
new icons

leiyre committed:
include example datasets

leiyre committed:
Merge branch 'feat/create-dataset-from-ui' of https://github.com/argilla-io/argilla into feat/create-dataset-from-ui

leiyre committed:
fix lint

leiyre committed:
allow white spaces in label and span text

leiyre committed:
lint fix

damianpumar committed:
🚑 Add validations

leiyre committed:
include validation messages in dialog

jfcalvo committed:
fix: add missing image type check

damianpumar committed:
🔥 Can map text questions

damianpumar committed:
Merge branch 'feat/create-dataset-from-ui' of https://github.com/argilla-io/argilla into feat/create-dataset-from-ui

frascuchon committed:
Merge branch 'develop' into feat/argilla-direct-feature-branch

damianpumar committed:
🐛 Show 409 error

leiyre committed:
include preview badge

leiyre committed:
include badge

leiyre committed:
update styles

leiyre committed:
Merge branch 'feat/create-dataset-from-ui' of https://github.com/argilla-io/argilla into feat/create-dataset-from-ui

leiyre committed:
update example dataset info

leiyre committed:
include styles to input

leiyre committed:
update colors

leiyre committed:
update home layout size

leiyre committed:
remove old datasets page

damianpumar committed:
🐛 Add metadata int8

damianpumar committed:
🚑 Supported new types.

frascuchon committed:
Merge branch 'develop' into feat/argilla-direct-feature-branch

frascuchon committed:
chore: API Call to fetch split stats

leiyre committed:
rename component

leiyre committed:
include links card for annotator view

leiyre committed:
Merge branch 'feat/create-dataset-from-ui' of https://github.com/argilla-io/argilla into feat/create-dataset-from-ui

leiyre committed:
fix validation and remove required for selector value

damianpumar committed:
🐛 Fix rating

leiyre committed:
add translations

leiyre committed:
Merge branch 'feat/create-dataset-from-ui' of https://github.com/argilla-io/argilla into feat/create-dataset-from-ui

leiyre committed:
include translations

leiyre committed:
include transaltions

leiyre committed:
remove key shortcuts in the form preview

leiyre committed:
remove unused component

leiyre committed:
uncommit

leiyre committed:
Merge branch 'feat/create-dataset-from-ui' of https://github.com/argilla-io/argilla into feat/create-dataset-from-ui

leiyre committed:
use copy code directive

leiyre committed:
fix import from python snippet

leiyre committed:
update badge styles

leiyre committed:
update card error

leiyre committed:
error styles for span

damianpumar committed:
🐛 Improve question validations

damianpumar committed:
Merge branch 'feat/create-dataset-from-ui' of https://github.com/argilla-io/argilla into feat/create-dataset-from-ui

leiyre committed:
improve validations for question cards

leiyre committed:
replace old dataset list route

leiyre committed:
show home card only for admir or owner

leiyre committed:
remove unused component

leiyre committed:
fix card button for dark mode

leiyre committed:
include placeholder

leiyre committed:
include placeholder translation

damianpumar committed:
✨ Fix selected split and imported from UI

damianpumar committed:
Merge branch 'feat/create-dataset-from-ui' of https://github.com/argilla-io/argilla into feat/create-dataset-from-ui

leiyre committed:
preselect first workspace and prevent to show selector if single workspace

leiyre committed:
update placeholder text

leiyre committed:
include warning for HF in snippet

leiyre committed:
improve styles

leiyre committed:
improve styles

leiyre committed:
fix condition

leiyre committed:
add dot

leiyre committed:
include error on load repo id

leiyre committed:
include es translations

leiyre committed:
improve responsive layout

nataliaElv committed:
Import from hub docs

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks
for more information, see https://pre-commit.ci

burtenshaw commented on argilla/docs/getting_started/quickstart.md:
````diff
@@ -59,6 +88,7 @@ pip install argilla
 ```
 
 ## Create your first dataset
````
We could actually just lose this section, and reference the guide on creating datasets in the SDK. 🤔

burtenshaw commented on argilla/docs/getting_started/quickstart.md:
```diff
@@ -133,8 +163,9 @@ dataset.records.log(records=data, mapping={"text": "review"})
 🎉 You have successfully created your first dataset with Argilla. You can now access it in the Argilla UI and start annotating the records.
 
 ## Next steps
+
```
We could have a next step to upload data to the hub, so that users could use this feature without code? Or maybe this should be higher up. 🤔

burtenshaw commented on argilla/docs/getting_started/quickstart.md:
```diff
@@ -48,7 +48,36 @@ In the sign in page:
+When you're happy with the result, you'll need to give a name to your dataset, select a workspace and choose a split, if applicable. Then, Argilla will start importing the dataset in the background. Now you're all set up to start annotating!
+
+!!! note
+    If your dataset is bigger than 10k records, at this stage Argilla will only import the first 10k. You can import the rest of the records at any point using the Python SDK.
```
```suggestion
    Argilla will only import the first 10k rows of a dataset. If your dataset is larger, you can import the rest of the records at any point using the Python SDK.
```

burtenshaw reviewed (commented):
Looks pretty good. 

This page is starting to get a lot of boxes. i.e. `!!! note` . When you read it on the frontend it's quite hard to follow. I would skim the page and see if some can be converted to in line text.

frascuchon committed:
[BUGFIX] argilla: support multi label values (#5625)

# Description
<!-- Please include a summary of the changes and the related issue.
Please also include relevant motivation and context. List any
dependencies that are required for this change. -->

This PR fixes problems when a class label sequence is mapped as
suggestions.

NOTE: This PR does not infer a sequence of class labels as multi-label
questions. Only prevent errors when a sequence of class labels column is
used as a suggestion (or other kinds of properties in argilla)

**Type of change**
<!-- Please delete options that are not relevant. Remember to title the
PR according to the type of change -->

- Bug fix (non-breaking change which fixes an issue)

**How Has This Been Tested**
<!-- Please add some reference about how your feature has been tested.
-->

**Checklist**
<!-- Please go over the list and make sure you've taken everything into
account -->

- I added relevant documentation
- I followed the style guidelines of this project
- I did a self-review of my code
- I made corresponding changes to the documentation
- I confirm My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature
works
- I have added relevant notes to the CHANGELOG.md file (See
https://keepachangelog.com/)

frascuchon committed:
Merge branch 'feat/argilla-direct-feature-branch' into feat/create-dataset-from-ui

nataliaElv committed:
Update argilla/docs/getting_started/quickstart.md

Co-authored-by: burtenshaw <ben@argilla.io>

leiyre committed:
update test

nataliaElv committed:
Refactor note blocks

leiyre committed:
update link to demo copy

leiyre committed:
include aria

sdiazlor reviewed (commented):
Just some comments as a person with little knowledge of this feature and is taking a first look.

- As the quickstart, I'd say to change the title of "Import a dataset from the Hub" to something more direct as it can be confusing for someone who doesn't know about Argilla. I'd say to use here "Create your first dataset", and that section remove it, adding just a note as Ben suggested and including SDK related stuff in a "Using the python SDK", similarly to "Deploy with Docker" where only it is mentioned and redirected. Or even combining the "importing long dataset" issue and the sdk further configuration/creation mention.
- I'll move the image after Fields and Questions so people can understand what they see.
- Not sure but regarding the Dataset Settings update, isn't it after giving it a name? I'd use a note or move it after that if that's the case.
- About workspaces, I'd keep the "About workspaces" as it seems it must be manually added and it's not automatically inferred?
- If this is the unique doc related to importing from the UI, it would be worthy to mention others like what about vectors or metadata, as those already knowing Argilla will have to check.

nataliaElv committed:
Merge branch 'feat/create-dataset-from-ui' into docs/ui-import-from-hub

nataliaElv committed:
Remove create dataset from SDK

nataliaElv committed:
Merge branch 'docs/ui-import-from-hub' of github.com:argilla-io/argilla into docs/ui-import-from-hub

frascuchon committed:
Merge branch 'releases/2.4.0' into docs/ui-import-from-hub

frascuchon commented on argilla/CHANGELOG.md:
```diff
@@ -27,6 +27,10 @@ These are the section headers that we use:
+### Removed
+
+- Removed name sanitizing for dataset settings names. This may cause issues with old server versions. Especially when working with `from_hub`. ([#5574](https://github.com/argilla-io/argilla/pull/5574))
+
```
```suggestion
```

frascuchon committed:
Update argilla/CHANGELOG.md

frascuchon committed:
Update useImportFromPython.ts

frascuchon committed:
ci: Configure docs permissions

frascuchon committed:
Merge branch 'releases/2.4.0' into docs/ui-import-from-hub

nataliaElv committed:
Final screenshot
